expansion/src/main/java/no/unit/nva/expansion/model/ExpandedParentPublication.java

```diff
@@ -65,13 +63,20 @@ private List<InputStream> getInputStreams(String publicationJsonString) {
 
     private Collection<? extends InputStream> fetchAll(List<URI> externalReferences) {
         return externalReferences.stream()
+                   .filter(this::isNotBlankUri)
                    .map(this::fetch)
-                   .flatMap(Optional::stream)
                    .map(IoUtils::stringToStream)
-                   .collect(Collectors.toList());
+                   .toList();
     }
 
-    private Optional<String> fetch(URI externalReference) {
-        return uriRetriever.getRawContent(externalReference, APPLICATION_JSON_LD.toString());
+    private boolean isNotBlankUri(URI uri) {
+        return StringUtils.isNotBlank(uri.toString());
+    }
+
+    private String fetch(URI externalReference) {
+        return uriRetriever.fetchResponse(externalReference, APPLICATION_JSON_LD.toString())
+                   .filter(i -> i.statusCode() / 100 == 2)
```

Comment:
Does cristin-proxy or channel-register return 30X? should we following redirect in that case?

Reply:
It's a good question. I have not done anything except make this fail yet.

Reply:
After consideration: If the requested URI redirects, then we must check that the redirect leads to a resource denoted by the originally requested URI. In our case: the root `@id` should be the same as the one we originally requested (this happens in some cases where a resource is requested with an accept header and a 303 redirect occurs to a particular serialization). Since we don't do this at any point, a 3XX error means we need to update our references.